### PR TITLE
#164489773 setup eslint

### PR DESCRIPTION
**Pull request by Inclet**
#### What does this PR do?
1. setup eslint : eslint, eslint-config-airbnb, eslint-plugin-import, eslint-plugin-jsx-a11y, eslint-plugin-react
#### Description of Task to be completed?
Have eslint in our project.
#### How should this be manually tested?
1. check `package.json` under `devdependencies`.
2. when cloning or pulling this branch do `npm install`
#### What are the relevant pivotal tracker stories?
Configure ESLint #164489773
[Pivotal Link](https://www.pivotaltracker.com/story/show/164489773)
#### Screenshots (if appropriate)
Screen Shot 2019-03-19 at 12 00 41

### CODE REVIEW

**Review by mwibutsa — APPROVED** [review]
Thank you for installing Eslint

**Comment by mystere10** [comment]
1. Remove the bracket in the PR name

**Review by frankhn — CHANGES_REQUESTED** [review]
Thanks for your effort @Inclet, but few things to note here.
 1. Description of the PR on the task to be completed isn't meaningful please check it out and make it more descriptive.
 2. Test for Eslint(Airbnb) should be tested when someone does not follow the rules specified respectively, so having Eslint in our project doesn't prove that it's working. I'd say that it would be better to dive into that and specify at least one thing that proves that Eslint really working.
     Specify the tests you run to verify your changes. 
     https://github.com/andela/engineering-playbook/blob/master/assets/pr-sample.png

3. Please add the rules in .eslintrc.js file as specified in the story
image

**Review by mystere10 — COMMENTED** [review]
Thank for the work done,
You should have mentioned that you used airbnb as eslint configuration in the description

**Comment by Inclet** [comment]
> Thanks for your effort @Inclet, but few things to note here.
> 
> 1. Naming the branch.
=> I named the branch `ch-configure-ESLINT-164489773` as you will see at the top of PR description.
> 2. Description of the PR on the task to be completed isn't meaningful please look check it out and make it more descriptive.
=> Check `How should this be manually tested?` part of the PR. 
> 3. We don't test the view, we test the action
=> Also Check `How should this be manually tested?` part of the PR.

**Review by frankhn — APPROVED** [review]

File: config/index.js [hunk 1]
```diff
@@ -1,4 +1,3 @@
 module.exports = {
-    secret:
-        process.env.NODE_ENV === "production" ? process.env.SECRET : "secret"
+  secret: process.env.NODE_ENV === 'production' ? process.env.SECRET : 'secret'
```
Comment (cyrielo) [inline config/index.js:2]: You should also have a local .env file for development

File: index.js [hunk 2]
```diff
@@ -1,93 +1,92 @@
...
 
 app.use(
-    session({
-        secret: "authorshaven",
-        cookie: { maxAge: 60000 },
-        resave: false,
-        saveUninitialized: false
-    })
+  session({
+    secret: 'authorshaven',
```
Comment (cyrielo) [inline index.js:22]: move this to the environment variable

**Review by cyrielo — CHANGES_REQUESTED** [review]
LGTM, see comments
PS: you should have a `.env` file to store your environment variables. Also, try to create a .env.sample file that would contain example values for those variables. only the ` .env.sample ` file should be committed and not the main .env file.

File: .env [hunk 3]
```diff
@@ -0,0 +1 @@
+secretkey = "NewNewThings"
```
Comment (cyrielo) [inline .env]: Remove this file from the commit and ignore the file using the `.gitignore` file. This file should be present in your folder but it should not be committed.

File: .gitignore [hunk 4]
```diff
@@ -5,6 +5,9 @@ logs
 
 npm-debug.log*
 
+#Environment Variables
+.env
```
Comment (cyrielo) [inline .gitignore:9]: Great, I see you're ignoring it here

File: config/index.js (hunk 1 above)
Comment (cyrielo) [reply config/index.js:2]: `secret: process.env.SECRET` should be enough

Comment (cyrielo) [reply config/index.js:2]:
```
secret: (process.env.NODE_ENV === 'production') ?  process:  process.env.DEVELOPMENT_KEY
```
OR

```
module.exports = {
  develop: {
      secret: process.env.DEVELOP_SECRET
   },
  production: {
    secret: process.env.PRODUCTION_SECRET
  }
}
```

**Pull request merged**